### ISSUE TITLE
FMS2 field size behaviour differing from FMS1

### ISSUE THREAD

**Issue by angus-g** [mom-ocean/MOM6#1531]
It looks like FMS2 has differing behaviour to FMS1 when given a field that is not 4D. For example, with the following file:
```cdf
netcdf forcing_obc {
dimensions:
        time = UNLIMITED ; // (365 currently)
        ny_segment_001 = 1 ;
        nx_segment_001 = 7201 ;
variables:
        double eta_t_segment_001(time, ny_segment_001, nx_segment_001) ;
}
```

the field sizes are reported as:
```
 fms1 get_field_size:        7201           1           1         365
 fms2 get_field_size:        7201           1         365           0
```

This particularly comes up in the open boundary code https://github.com/NOAA-GFDL/MOM6/blob/037af8ec782fead0f7aebff60a5a9e4406930d55/src/core/MOM_open_boundary.F90#L866-L897 which assumes the third dimension of the size array always corresponds to depth.

Maybe this is something that should be dealt with within FMS itself, but there are also the infra and OBC layers involved here.

**Comment by MJHarrison-GFDL** [comment]
We can retain the original FMS1 behavior by introducing an optional flag in MOM_io_infra:get_field_size  to set whether the newer FMS2 behavior is desired, or the original FMS1 behavior of returning the 4-D shape in x/y/z/t order.  We can set the default to FMS1 style, which would involve fewer changes to the code (although it is only called from a handful of places) and newer codes can use this flag as desired.

**Comment by marshallward** [comment]
I believe this has been resolved by NOAA-GFDL/MOM6#111.  We can reopen if it comes back.
